djautotask/sync_rest.py

```diff
@@ -181,6 +181,9 @@ def persist_page(self, records, results):
                     results.updated_count += 1
                 else:
                     results.skipped_count += 1
+            except TypeError as e:
+                logger.warning('{}'.format(e))
+                continue
```

Comment:
This can be removed, it is impossible for this to happen. This is running for each element in `records` so we know it won't be `None`. The problem only exists in `fetch_sync_by_id`.

Reply:
I tested the following code snippet. Yes, Sentry doesn't complain in this code, but when AT-side is unstable (for example, unexpected errors such as seen at `fetch_sync_by_id` or max-request occurrences, and so on), it could be some kind of safety net.

![image](https://user-images.githubusercontent.com/73486314/134751937-236a65a9-b404-4166-a56d-abca41279886.png)

But, I can agree with you because we haven't seen any errors yet here.

Reply:
Also I don't think the AT-side being unstable is guaranteed to be a TypeError, is it? We shouldn't just guess at exceptions.

Reply:
Okay, I will remove that.